### PR TITLE
Switch off logging

### PR DESCRIPTION
**Pull request by corinnebosley**
This is for reference only.  In actual fact, this doesn't seem to make any functional difference.

### CODE REVIEW

**Comment by marqh** [comment]
> This is for reference only. In actual fact, this doesn't seem to make any functional difference.

@corinnebosley 
the case for this statement would be enhanced by fixing the coding standards test failure.
I would like to see this passing this travis tests, i think it is worth discussing the benefit/cost of this logging implementation

do any developers know of where this is on use?
answers to me on a post card please

File: lib/iris/fileformats/rules.py [hunk 1]
```diff
@@ -209,17 +142,6 @@ def _disable_deprecation_warnings():
         _enable_rules_deprecations = old_flag_value
 
 
-# Defines the "log" function for this module
-# This is a 'private' version:  The public one is now deprecated (see on).
-_log_rules = _prepare_rule_logger()
-
-
-# Provide a public 'log' function, which issues a deprecation warning.
-def log(*args, **kwargs):
```
Comment (pelson) [inline lib/iris/fileformats/rules.py]: Public API. Removal will need to be documented.

**Comment by corinnebosley** [comment]
@pelson I want this merged please.  Waiting for tests, should only be pep8 problems but I will try and make sure this is all sorted by the end of today.

Also, don't mean to sound bossy, it's just my nature...

File: lib/iris/fileformats/rules.py [hunk 2]
```diff
@@ -24,6 +24,13 @@
 import six
 
 import collections
+from contextlib import contextmanager
```
Comment (pelson) [inline lib/iris/fileformats/rules.py]: ?  I'm confused by these ones...

Comment (corinnebosley) [reply lib/iris/fileformats/rules.py]: Me too.  I think that must be conflict confusion.  I'll just take that bit out.

**Comment by pelson** [comment]
Seems that there may actually be a genuine error, but don't understand how it relates to this PR...

```
======================================================================
FAIL: test_concatenate_cell_measure_match (iris.tests.integration.test_netcdf.TestCellMeasures)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-2.0a0-py2.7.egg/iris/tests/integration/test_netcdf.py", line 258, in test_concatenate_cell_measure_match
    self.assertEqual(len(cubes), 1)
AssertionError: 2 != 1
----------------------------------------------------------------------
```

**Comment by pelson** [comment]
Going to need to add this to the what's new as an incompatible change

**Pull request merged**